mongo/index_view.go

```diff
@@ -435,6 +447,15 @@ func (iv IndexView) DropOne(ctx context.Context, name string, opts ...*options.D
 	return iv.drop(ctx, name, opts...)
 }
 
+// DropKeyOne executes a dropIndexes operation to drop an index on the collection. If the operation succeeds, this returns
+// a BSON document in the form {nIndexesWas: <int32>}. The "nIndexesWas" field in the response contains the number of
+// indexes that existed prior to the drop.
+//
+// The key parameter should be the keySpecDocument of the index to drop.
+func (iv IndexView) DropKeyOne(ctx context.Context, key bsoncore.Document, opts ...*options.DropIndexesOptions) (bson.Raw, error) {
```

Comment:
We should not expose unstable API, such as bsoncore.Document, to users in the stable API. This is because unstable API is subject to change and such changes realized in the stable API would be breaking. Update the function signature to use interface{} as the type for the `key` parameter.